server/svix-server/src/v1/endpoints/endpoint/mod.rs

```diff
@@ -262,6 +274,66 @@ impl ModelIn for EndpointHeadersPatchIn {
     }
 }
 
+#[derive(Deserialize, Serialize)]
+pub struct EndpointStatsOut {
+    success: i64,
+    pending: i64,
+    sending: i64,
+    fail: i64,
+}
+
+#[derive(Debug, FromQueryResult)]
+pub struct EndpointStatsQueryOut {
+    status: MessageStatus,
+    count: i64,
+}
+
+async fn endpoint_stats(
+    Extension(ref db): Extension<DatabaseConnection>,
+    Path((_app_id, endp_id)): Path<(ApplicationIdOrUid, EndpointIdOrUid)>,
+    AuthenticatedApplication {
+        permissions: _,
+        app,
+    }: AuthenticatedApplication,
+) -> crate::error::Result<Json<EndpointStatsOut>> {
+    if let Some(endpoint) =
+        crate::db::models::endpoint::Entity::secure_find_by_id_or_uid(app.id, endp_id)
+            .one(db)
+            .await?
+    {
+        let query_out: Vec<EndpointStatsQueryOut> =
+            messagedestination::Entity::secure_find_by_endpoint(endpoint.id)
+                .select_only()
+                .column(messagedestination::Column::Status)
+                .column_as(messagedestination::Column::Status.count(), "count")
+                .group_by(messagedestination::Column::Status)
+                .filter(
+                    messagedestination::Column::CreatedAt
+                        .gte(chrono::Utc::now() - chrono::Duration::days(28)),
+                )
+                .into_model::<EndpointStatsQueryOut>()
+                .all(db)
+                .await
+                .map_err(|e| {
+                    tracing::error!("ERROR {}", e);
+                    e
+                })?;
+        let mut query_out = query_out
+            .into_iter()
+            .map(|EndpointStatsQueryOut { status, count }| (status, count))
+            .collect::<HashMap<_, _>>();
+
+        Ok(Json(EndpointStatsOut {
+            success: query_out.remove(&MessageStatus::Success).unwrap_or(0),
+            pending: query_out.remove(&MessageStatus::Pending).unwrap_or(0),
+            fail: query_out.remove(&MessageStatus::Fail).unwrap_or(0),
+            sending: query_out.remove(&MessageStatus::Sending).unwrap_or(0),
+        }))
+    } else {
+        Err(HttpError::unauthorized(None, None).into())
```

Comment:
No need to put the whole body in an if/else, just do the fetch/return-on-failure short-circuit like is done literally everywhere else.

Also, the error should be `not_found` not `unauthorized`...